lib/file.js

```diff
@@ -358,26 +358,28 @@ class File extends EventEmitter {
     // https://mega.nz/file/file_handler#file_key
     // https://mega.nz/folder/folder_handler
     // https://mega.nz/folder/folder_handler#folder_key
+    // https://mega.nz/folder/folder_handler#folder_key/file/file_handler
 
     const url = parse(opt)
     if (url.hostname !== 'mega.nz' && url.hostname !== 'mega.co.nz') { throw Error('Invalid URL: wrong hostname') }
     if (!url.hash) throw Error('Invalid URL: no hash')
 
     if (url.path.match(/\/(file|folder)\//) !== null) {
       // new format
+      const split = url.hash.substr(1).split('/file/')
       const fileHandker = url.path.substring(
         url.path.lastIndexOf('/') + 1,
         url.path.length + 1
       )
-      const fileKey = url.hash.substr(1)
+      const fileKey = split[0].substr(1)
```

Comment:
`.substr(1)` is being called twice.

```suggestion
      const fileKey = split[0]
```